### ISSUE TITLE
[XBOX ONE] Contains 3.16.0 - Channels not loading

### ISSUE THREAD

**Issue by dvdbot** [kodi-pvr/pvr.vuplus#167]
Like this issue https://github.com/kodi-pvr/pvr.vuplus/issues/161 I've got the problem of channels not loading. The main problem is, I can't update the PVR Addon on there. Any workarround known?

Also the two API urls:
```
http://192.168.0.10/api/getservices?provider=1&picon=1&sRef=1%3A7%3A1%3A0%3A0%3A0%3A0%3A0%3A0%3A0%3AFROM%20BOUQUET%20%22userbouquet%2Evh01%2Etv%22%20ORDER%20BY%20bouquet%27
```
```
http://192.168.0.10/api/getservices?provider=1&picon=1&sRef=1%3A7%3A1%3A0%3A0%3A0%3A0%3A0%3A0%3A0%3AFROM%20BOUQUET%20%22userbouquet%2Evh01%2Etv%22%20ORDER%20BY%20bouquet
```
return both the following
```json
{"services": [], "pos": 0}
```

**Comment by phunkyfish** [comment]
Are those URLs from your own log?

Can you create a full debug log from restart?

I think maybe the OpenWebIf if version needs to be increased for the api calls to work. It’s difficult to tell as the releases are not tagged in the OpenWebIf repo.

**Comment by phunkyfish** [comment]
For the API calls they need to be from your log as they use a channel reference for your STB.

**Comment by dvdbot** [comment]
I can't really access the logs from my pc as for as I know... can't access the Kodi running on the Xbox One as a network share and already tried to copy the logs using the File Manager but No copy option

I'll try to get my hands on the logs

**Comment by dvdbot** [comment]
The exact same piece is also shown during startup in the log

**Comment by phunkyfish** [comment]
So now all you need to do is enable debug logging, restart kodi and post that log.

**Comment by phunkyfish** [comment]
The link above shows you how to enable debug logging.

**Comment by dvdbot** [comment]
Oh duh!

**Comment by dvdbot** [comment]
New log :P
https://paste.kodi.tv/afinuhohag

**Comment by phunkyfish** [comment]
Ok, so now open this URL  in a browser and post the result using https://paste.kodi.tv

```
http://192.168.0.10:80/api/getservices?provider=1&picon=1&sRef=1%3A7%3A1%3A0%3A0%3A0%3A0%3A0%3A0%3A0%3AFROM%20BOUQUET%20%22userbouquet%2EZiggo%2Etv%22%20ORDER%20BY%20bouquet
```

**Comment by phunkyfish** [comment]
So I’ve tested this and the fix works with the JSON output.

I’ve also noticed my original assumption that the full JSON API was available since version 1.3.0 of OpenWebIf is incorrect, it needs to be version 1.3.5

**Comment by phunkyfish** [comment]
Duplicate of #161

**Comment by dvdbot** [comment]
So I need to update my OpenWebIf?

**Comment by dvdbot** [comment]
Cause I can't update the kodi add-on on the Xbox One

**Comment by phunkyfish** [comment]
So the issue will be fixed in version 3.16.1 once the PR is approved.

As a workaround you could try upgrading to OpenWebIf 1.3.5 on your STB.

**Comment by phunkyfish** [comment]
Why can’t you update the addon on the Xbox?

**Comment by phunkyfish** [comment]
Have you tried to update it and instead choose 3.15.4?

**Comment by dvdbot** [comment]
I tried to press the update button but it says `There are no updates available` - I believe on the Xbox One currently the binary addons and some others are shipped with Kodi itself - not pulled from Repository

**Comment by dvdbot** [comment]
I can try to update the OpenWebIf - if I can find how 😂

**Comment by phunkyfish** [comment]
No problem, anytime
